controllers/front/success.php

```diff
@@ -47,19 +49,57 @@ public function postProcess()
 
             Tools::redirect($redirectLink);
         }
+        try {
+            $authResponseBody = $this->assertTransaction($orderId);
 
-        $orderLink = $this->context->link->getPageLink(
-            'order-confirmation',
-            true,
-            null,
-            [
-                'id_cart' => $cartId,
-                'id_module' => $moduleId,
-                'id_order' => $orderId,
-                'key' => $secureKey,
-            ]
-        );
+            if ($authResponseBody->getTransaction()->getStatus() === TransactionStatus::CANCELED) {
+                throw new Exception('Failed to authorize transaction');
+            }
 
-        Tools::redirect($orderLink);
+            Tools::redirect($this->context->link->getPageLink(
+                'order-confirmation',
+                true,
+                null,
+                [
+                    'id_cart' => $cartId,
+                    'id_module' => $moduleId,
+                    'id_order' => $orderId,
+                    'key' => $secureKey,
+                ]
+            ));
+        } catch (Exception $e) {
+            PrestaShopLogger::addLog(
+                sprintf(
+                    '%s has caught an error: %s',
```

Comment:
You don't need to specify "has caught an error" and stuff. Just write "Failed to authorize transaction". 
Also, if status is cancelled, does it really mean authorization failed?